tagstudio/src/core/library.py

```diff
@@ -311,6 +315,48 @@ def compressed_dict(self) -> JsonCollation:
         return obj
 
 
+class IsDirModifiedHandler(FileSystemEventHandler):
+    def __init__(self, lib: "Library", callback: Callable[[], None]) -> None:
+        self.library: Library = lib
+        self.callback: Callable[[], None] = callback
+        super().__init__()
+
+    @cached(cache=LRUCache(maxsize=32), key=lambda _, path, time: (path, time))
+    def event_helper(self, filepath: str, time: int) -> None:
+        _ = time
+        # Here update internal vars with single new file
+        if "$RECYCLE.BIN" in filepath or "tagstudio_thumbs" in filepath:
+            return
+
+        assert not os.path.isdir(filepath)
```

Comment:
I think this should not be an assertion because its not essential to keep running and should just be handled somewhere else